pyodide_build/config.py

```diff
@@ -13,8 +14,81 @@
 
 
 class ConfigManager:
+    """
+    Configuration manager for pyodide-build.
+    This class works "before" installing the cross build environment.
+    So it does not have access to the variables that are retrieved from the cross build environment.
```

Comment:
The new config variable needs to be accessed before downloading the cross-build env, so I subclassed the `ConfigManager` into `ConfigManager` and `CrossBuildEnvConfigManager`. The former can be used even before installing the cross build env.

Reply:
Can we separate splitting these into its own PR?

Reply:
Thanks. Sounds reasonable. Let me split the PR